.github/workflows/release.yaml

```diff
@@ -0,0 +1,87 @@
+# Makes an automated release
+#
+# To trigger this workflow, ensure the CHANGELOG is up to date, then push the commit
+# to be released to a `release/*` branch.
+#
+# e.g.
+#     git checkout -b release/v1.0.0
+#     git push
+#
+name: Release
+
+on:
+  push:
+    branches: [release/*]
+
+jobs:
+  create-release:
+    name: Create GitHub Release and tag
+    runs-on: ubuntu-latest
+    environment: Release
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cucumber-actions/create-release@v1.1.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+  publish-rubygem:
+    name: Publish Ruby Gem
+    needs: create-release
+    runs-on: ubuntu-latest
+    environment: Release
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.0.2
+          bundler-cache: true
+      - uses: cucumber-actions/publish-rubygem@v1.0.0
+        with:
+          rubygems_api_key: ${{ secrets.RUBYGEMS_API_KEY }}
+          working_directory: ruby
+
+  publish-npm:
+    name: Publish NPM package
+    needs: create-release
+    runs-on: ubuntu-latest
+    environment: Release
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+      - uses: cucumber-actions/publish-npm@v1.0.0
+        with:
+          npm-token: ${{ secrets.NPM_TOKEN }}
+          working-directory: javascript
+
+  publish-java:
+    name: Publish Maven package
+    needs: create-release
+    runs-on: ubuntu-latest
+    environment: Release
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '11'
+
+      - uses: samuelmeuli/action-maven-publish@v1
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          gpg_passphrase: ${{ secrets.GPG_PASSPHRASE }}
+          nexus_username: cukebot
+          nexus_password: ${{ secrets.SONATYPE_PASSWORD }}
+          maven_profiles: sign-source-javadoc
+          directory: java
+
+  publish-go:
+    name: Create go/* tag
+    needs: create-release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create go/* tag
+        run: |-
+          git config user.name "cukebot"
+          git config user.email "actions@users.noreply.github.com"
+          git tag "go/${{ steps.create-release.outputs.version }}"
```

Comment:
We could also use the https://github.com/cucumber-actions/versions action as a prior step in this workflow.

Or, if we factor out our own action for this we can make it self-sufficient about looking up the version.